src/crimson/common/errorator.h

```diff
@@ -1147,7 +1148,7 @@ namespace ct_error {
   using enametoolong = ct_error_code<static_cast<int>(std::errc::filename_too_long)>;
   using eexist = ct_error_code<static_cast<int>(std::errc::file_exists)>;
   using edquot = ct_error_code<int(122)>;
-  using cmp_fail = ct_error_code<int(4095)>;
+  using cmp_fail = ct_error_code<int(MAX_ERRNO)>;
```

Comment:
Let's not use MAX_ERRNO for this.  Define a

```
constexpr int cmp_fail_error_value = 4095;
```

and check for that instead of MAX_ERRNO.  That will give anyone reading the code a hint of where to look for an explanation.